### DISCUSSION TITLE
vercel.json -> vercel.config.js

### DISCUSSION THREAD

**Discussion by qdozaq** [vercel/community#25] [Idea]
Any plans to enable us to use a js config for the `vercel.json`?
Essentially the same as a `next.config.js`

For my use case, this would be incredibly helpful when trying rewrite multiple projects to one domain while keeping their branches in sync:

```js
const { VERCEL_GIT_COMMIT_REF } = process.env

module.exports = {
  async rewrites() {
    return [
      {
        source: "/",
        destination: `https://home-${VERCEL_GIT_COMMIT_REF}-git-doza.vercel.app`,
      },
      {
        source: "/dashboard",
        destination: `https://dashboard-${VERCEL_GIT_COMMIT_REF}-git-doza.vercel.app`,
      },
    ]
  },
}

```

My solution right now is to just use an empty `next.js` project and using the `next.config.js` to handle these kinds of rewrites, though something more succinct would be nice.

**Comment by Pieparker** [comment]
Hey there: no plans as far as I'm aware so thank you for creating this feature suggestion!

**Reply by qdozaq** [reply]
Sure thing!

**Comment by abernh** [comment]
Yes, that would come in very handy for a SSG Nuxt project.
Via feature-flags I'm en-/disabling some pages and now it would be nice if I could feed those flags also into a `vercel.config.js` to auto-reroute disabled routes for propper `302` http status codes.
